src/wp-includes/rest-api.php

```diff
@@ -1284,12 +1285,36 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 			return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s is not of type %2$s.' ), $param, 'object' ) );
 		}
 
+		/*
+		 * @ticket 48818
+		 */
+		if ( isset( $args['properties'] ) && ! empty( $value ) ) {
+			if ( isset( $args['required'] ) ) { // schema version 4
+				foreach ( $args['required'] as $name ) {
+					if ( ! isset( $value[ $name ] ) ) {
+						/* translators: %s: Property of an object. */
+						return new WP_Error( 'rest_property_required', sprintf( __( '%1$s is a required property of Object.' ), $name ) );
+					}
+				}
+			} else { // schema version 3
+				foreach ( $args['properties'] as $name => $property ) {
+					if ( isset( $property['required'] ) && true === (bool) $property['required'] && ! isset( $value[ $name ] ) ) {
+						/* translators: %s: Property of an object. */
+						return new WP_Error( 'rest_property_required', sprintf( __( '%1$s is a required property of Object.' ), $name ) );
+					}
+				}
+			}
+		}
+
 		foreach ( $value as $property => $v ) {
 			if ( isset( $args['properties'][ $property ] ) ) {
 				$is_valid = rest_validate_value_from_schema( $v, $args['properties'][ $property ], $param . '[' . $property . ']' );
 				if ( is_wp_error( $is_valid ) ) {
 					return $is_valid;
 				}
+			} elseif ( isset( $args['properties'][ $property ]['required'] ) && $args['properties'][ $property ]['required'] ) {
```

Comment:
What is the purpose of this check? We're looping over the actual properties of the passed value, and checking against the schema for each property.

Reply:
Can't remember and removed